.gitlab-ci.yml

```diff
@@ -809,9 +810,26 @@ dev_master_quay:
     - inv -e docker.publish ${SRC_DSD}:${SRC_TAG} quay.io/datawhale/dogstatsd-dev:master
 
 #
-# deploy
+# Check Deploy
 #
 
+# Check that the current version hasn't already been deployed (we don't want to
+# overwrite a public package). To update a erroneous package, first remove it
+# from our S3 bucket.
+check_already_deployed_version:
+  <<: *run_when_triggered
+  stage: check_deploy
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/deploy:latest
+  before_script:
+    - ls $OMNIBUS_PACKAGE_DIR
+  tags: [ "runner:main", "size:large" ]
+  script:
+    - set +x
```

Comment:
I kind of feel like this `set +x` should be in the script? I'm also not sure if this would carry over to the shell the script is executed in?